docs/src/main/sphinx/security/cert-auth.rst

```diff
@@ -0,0 +1,108 @@
+==========================
+Certificate authentication
+==========================
+
+Trino can be configured to support certified clients seeking validation from a
+Trino server. This authentication method is only provided to support sites that
+have an absolute requirement for client authentication *and already have* client
+certificates for each client. Sites in this category have an existing PKI
+infrastructure, possibly including an onsite CA.
+
+This feature is *not* appropriate for sites that would need to generate a set of
+client certificates in order to use this authentication method. Starburst
+recommends instead configuring one of the password-based authentication methods,
+including :doc:`password-file`, :doc:`ldap`, and :doc:`salesforce`; or choosing
+an :doc:`OAuth2 <oauth2>`-compatible authentication vendor.
+
+
+Using certificate authentication
+--------------------------------
+
+With certificate authentication enabled, the validation sequence is the
+following:
+
+* A client verfies the coordinator's certificate, and makes an attempt to
+  connect to the coordinator.
+* The coordinator asks the client for its certificate.
+* The client responds with *its* certificate, validated by *its* trust store.
```

Comment:
that trust store is on the client machine right?

Reply:
No. It's the standard system trust store available to the coordinator normally, and does not need to be the same trust store that the coordinator uses to validate its own server certs (but probably is the same). Remember that cert auth only works on sites that "already have client certs for each client", for which the network admins must have taken steps to be validated everywhere inside their network. That means either that the client certs are valid against CAs always found in the standard Java trust store, or that the trust store on every system has been augmented by the network admins with a custom CA, or that the site runs a custom internal CA service that every client points to.

All of this is presumed in the words "already have client certs for each client." Or do you think we should spell this out in this doc?

Reply:
Made it much clearer whose trust store validates which certs, and that they do not need to be the same cert for server and client.